### PR TITLE
Provide FEDERATIONS_DOMAIN_MAP environment variable to federation tests.

### PR DESCRIPTION
**Pull request by madhusudancs**
cc @quinton-hoole @nikhiljindal @mml

### CODE REVIEW

**Comment by madhusudancs** [comment]
Who should I ping for review and merge?

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml [hunk 1]
```diff
@@ -72,6 +72,7 @@
               export PROJECT="k8s-jkns-e2e-gce-federation"
               export FEDERATION=true
               export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-gce-federation"
+              export FEDERATIONS_DOMAIN_MAP="federation=ubernetes-test.com"
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml]: is ubernetes-test.com a real domain? should it be?

Comment (quinton-hoole) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml]: It is not a domain that we own.  It is probably better to use a RFC 2606-compliant domain name here (and omit the "ubernetes" part, as we're trying to discourage the use of that slang term in favor of "Cluster Federation".

https://tools.ietf.org/html/rfc2606#page-2

e.g. 

$ gcloud dns managed-zones create federation-dot-test --description "Test domain for Kubernetes cluster federation" --dns-name "federation.test"
Created [.../managedZones/federation-dot-test].
NAME                 DNS_NAME          DESCRIPTION
federation-dot-test  federation.test.  Test domain for Kubernetes cluster federation

**Comment by quinton-hoole** [comment]
I think that this is superceded by #27986 anyway, right @madhusudancs ?

**Comment by nikhiljindal** [comment]
No. We are not defaulting FEDERATION_NAME and DNS_ZONE_NAME in #27986, so will need this PR.

**Comment by ixdy** [comment]
Another question I have: do you need to set `FEDERATIONS_DOMAIN_MAP` for both build and test? I'm not sure where it's used, but that's unlike most other env vars.

**Comment by quinton-hoole** [comment]
Agreed @ixdy , Madhu and I just discussed this and I think we agreed that we only need it in -up and -test, but not in -build.  Either way, ignore this PR for now.  We will figure all of this out after 1.3.

**Comment by madhusudancs** [comment]
@ixdy @quinton-hoole `ubernetes-test.com` is a domain name that we own. I registered the domain and made it point to Google Cloud DNS.

**Comment by madhusudancs** [comment]
@ixdy `FEDERATIONS_DOMAIN_MAP` is not required for running the tests. It is required for building the binaries. I wasn't sure whether the binaries were built as part of the test job.

**Comment by nikhiljindal** [comment]
As discussed with @quinton-hoole we can change the domain later if required. We want to get this in to get the tests passing on jenkins.

**Comment by madhusudancs** [comment]
@nikhiljindal @ixdy sent a fix to remove the environment variable from the test job.

**Comment by madhusudancs** [comment]
@ixdy This PR is in its desired shape now. PTAL.

**Comment by madhusudancs** [comment]
`k8s-federation.com` is a domain we own.

**Comment by ixdy** [comment]
LGTM. merge at will.

**Comment by madhusudancs** [comment]
@ixdy thanks!

**Pull request merged**